test/server/core/abilities/keyword/Shielded.spec.js

```diff
@@ -0,0 +1,21 @@
+describe('Shielded keyword', function() {
+    integration(function() {
+        describe('When a unit with the Shielded keyword', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['crafty-smuggler']
+                    }
+                });
+            });
+
+            it('enters play, it receives a shield', function () {
+                this.player1.clickCard(this.craftySmuggler);
+                expect(this.craftySmuggler.upgrades.length).toBe(1);
+                expect(this.craftySmuggler.upgrades[0].title).toBe('Shield');
```

Comment:
Can simplify these two expectations to be `expect(this.craftySmuggler).toHaveExactUpgradeNames(['shield']);`

Reply:
Ahh right, it's in state watchers. I hate having adhd lol. I just merged that in so it should be available now.